src/app/teacher-round.service.spec.ts

```diff
@@ -15,15 +17,35 @@ describe('TeacherRoundService', () => {
   };
 
   beforeEach(() => {
+    const mockCurrentRoundPath$ = new BehaviorSubject<RoundPath>(null);
+    const mockSessionId$ = new BehaviorSubject<string>(fakeSessionId);
+
+    const mockTeacherSessionService: Partial<TeacherSessionService> = {
+      currentRoundPath$: mockCurrentRoundPath$,
+      sessionId$: mockSessionId$,
+    };
+
     const mockFirebaseService = jasmine.createSpyObj<Partial<FirebaseService>>(
       'firebaseService',
-      ['updateRoundData', 'setRoundData'],
+      ['updateRoundData', 'setRoundData', 'createRoundInSession', 'setCurrentRound'],
     );
 
+    mockFirebaseService.createRoundInSession.and.callFake(() => {
+      const fakeRoundPath = {sessionId: fakeSessionId, roundId: 'demo-round'};
+      mockCurrentRoundPath$.next(fakeRoundPath);
```

Comment:
@helloworld12321 Is there a reason you set the mockCurrentRoundPath$ here? The actual createRoundInSession in FirebaseService does not change currentRoundPath$

Reply:
Ah, good catch, thanks! I'll check that out later today.

Reply:
Fixed!